src/main/java/htsjdk/samtools/BAMFileReader.java

```diff
@@ -763,7 +801,7 @@ private void assertIntervalsOptimized(final QueryInterval[] intervals) {
     /**
      * Iterate over the SAMRecords defined by the sections of the file described in the ctor argument.
      */
-    private class BAMFileIndexIterator extends BAMFileIterator {
+    public class BAMFileIndexIterator extends BAMFileIterator {
```

Comment:
Why does this need to be made public? It seems like there are no direct usages in any downstream PRs